xla/service/gpu/gemm_rewriter.cc

```diff
@@ -1374,6 +1374,15 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
 
     std::unique_ptr<HloInstruction> new_gemm =
         existing_gemm->CloneWithNewShape(instr->shape());
+
+    // The F8ConvertD may change the output dtype. We need to turn off the
+    // output to operand aliasing when it happens.
+    const auto& maybe_bias = existing_gemm->operand(2);
```

Comment:
Is there a good reason for using auto here? Since this is only used to access the shape, maybe have a Shape variable instead or remove it?